gensim/corpora/dictionary.py

```diff
@@ -193,6 +195,12 @@ def filter_extremes(self, no_below=5, no_above=0.5, keep_n=100000):
         good_ids = (
             v for v in itervalues(self.token2id)
             if no_below <= self.dfs.get(v, 0) <= no_above_abs)
+        # add ids of keep_tokens elements to good_ids
+        if keep_tokens:
+            keep_ids = [self.token2id[v] for v in keep_tokens if v in self.token2id]
+            good_ids_copy =  (v for v in itervalues(self.token2id) if no_below <= self.dfs.get(v, 0) <= no_above_abs)
```

Comment:
please keep to a single iteration of `token2id` by adding a ` or v in keep_tokens` check if `keep_tokens` is present